### PR TITLE
Convenience flag to query stake distribution with `drep-state` query

### LINKED ISSUE #551: [FR] - include the stake-distribution within the drep-state query
To get the whole information about a drep keyHash, we have to make two queries:
1. query the `drep-state`
```json
[
    [
        {
            "keyHash": "2c0e9f599655bd3f7f313a3a61fe05fb9caac4cc3d1aa77740fc9579"
        },
        {
            "anchor": null,
            "deposit": 2000000,
            "expiry": 223
        }
    ]
]
```
2. query the `drep-stake-distribution`
```json
[
    [
        "drep-keyHash-2c0e9f599655bd3f7f313a3a61fe05fb9caac4cc3d1aa77740fc9579",
        1625348
    ]
]
```

Would it be possible to directly have an entry for the currently delegated amount within the `drep-state` output?

Like:
```json
[
    [
        {
            "keyHash": "2c0e9f599655bd3f7f313a3a61fe05fb9caac4cc3d1aa77740fc9579"
        },
        {
            "anchor": null,
            "deposit": 2000000,
            "expiry": 223
        },
        {
            "delegation": 1625348
        }
    ]
]
```
or similar?

### PR DESCRIPTION
**Pull request by carlhammann**
# Changelog

```yaml
- description: |
    Add `--include-stake` flag to obtain the stake in the `drep-state` query
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

This PR closes #551. 

I introduce an overlap between the `drep-state` and `drep-stake-distribution` queries. With the `--include-stake` flag, the former now subsumes the latter. AFAICT, the queries are exposed separately because [`drep-stake-distribution` is expensive](https://github.com/IntersectMBO/ouroboros-consensus/blob/404aaec8a4c8091594b4bc5a38e26f6ea9627f6b/ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs#L254). This makes obtaining all information associated with a DRep credential a two-command process. With the new flag, the expensive query is still guarded, but the CLI gets more user-friendly.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Query.hs [hunk 1]
```diff
@@ -323,6 +323,16 @@ pQueryDRepStateCmd era envCli = do
         <*> pConsensusModeParams
         <*> pNetworkId envCli
         <*> many pDRepVerificationKeyOrHashOrFile
+        <*> Opt.flag True False (mconcat
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Query.hs]: Implement a sum type for this to avoid boolean blindness.

**Review by Jimbo4350 — COMMENTED** [review]
LGTM but one comment

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Query.hs (hunk 1 above)
Comment (carlhammann) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Query.hs]: Good point, done!

**Comment by gitmachtl** [comment]
looks like the current command defaults to `--include-stake`, i guess that should be false per default?

also, the generated output does not contain the keyhash anymore. looks like:

```json
[
    {
        "anchor": {
            "dataHash": "1111111111111111111111111111111111111111111111111111111111111111",
            "url": "https://bit.ly/3zCH2HL"
        },
        "deposit": 2000000,
        "expiry": 188,
        "stake": null
    },
    {
        "anchor": null,
        "deposit": 2000000,
        "expiry": 230,
        "stake": null
    },
    {
        "anchor": null,
        "deposit": 2000000,
        "expiry": 216,
        "stake": null
    },
    {
        "anchor": {
            "dataHash": "f8f21b94aef91b0d5e7ff330889d60f665529897b5c1ae4a1aa608b6249e20e6",
            "url": "https://github.com/Mborongoma/SANCH-GOV-TESTES/blob/main/bio.txt"
        },
        "deposit": 2000000,
        "expiry": 227,
        "stake": null
    },
    {
        "anchor": {
            "dataHash": "fd1b4e7fd844a8e2805bfda173f6461fd59812fcad4ce8a69b5ab427f7226cf7",
            "url": "https://my-ip.at/test/hwdrep.json"
        },
        "deposit": 2000000,
        "expiry": 232,
        "stake": null
    }
]
```

**Comment by carlhammann** [comment]
> looks like the current command defaults to `--include-stake`, i guess that should be false per default?

Well spotted, @gitmachtl! The flag is actually off by default, in the sense that the _query_ isn't performed, but the _output_ includes the `"stake"` field always. That's confusing, so I changed that behaviour in [f11dd55](https://github.com/IntersectMBO/cardano-cli/pull/557/commits/f11dd55d72e311fcbabd88c89feba7d3e014c8e5): Now the field should only be present when `--include-stake` is used.

**Comment by gitmachtl** [comment]
> > looks like the current command defaults to `--include-stake`, i guess that should be false per default?
> 
> Well spotted, @gitmachtl! The flag is actually off by default, in the sense that the _query_ isn't performed, but the _output_ includes the `"stake"` field always. That's confusing, so I changed that behaviour in [f11dd55](https://github.com/IntersectMBO/cardano-cli/pull/557/commits/f11dd55d72e311fcbabd88c89feba7d3e014c8e5): Now the field should only be present when `--include-stake` is used.

thx! and the keyHash is missing in the output too for some reason.

**Comment by carlhammann** [comment]
> thx! and the keyHash is missing in the output too for some reason.

Fixed in [9bd912b](https://github.com/IntersectMBO/cardano-cli/pull/557/commits/9bd912b6d8de006caf993a40e103d53ed6e97bf1). Thanks for the reminder :bow:

**Comment by gitmachtl** [comment]
hi, any plans to merge this into main soon? thx.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs [hunk 2]
```diff
@@ -1419,24 +1420,40 @@ runQueryDRepState
...
   writeOutput mOutFile $
-    second drepStateToJson <$> Map.assocs drepState
+    drepStateToJson drepStakeDistribution <$> Map.assocs drepState
   where
-    drepStateToJson ds = A.object
-      [ "expiry" .= (ds ^. Ledger.drepExpiryL)
-      , "anchor" .= (ds ^. Ledger.drepAnchorL)
-      , "deposit" .= (ds ^. Ledger.drepDepositL)
-      ]
+    drepStateToJson stakeDistr (cred, ds) = (cred,) . A.object $
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs:1460]: This is fine for now but we should convert this into a `ToJSON` instance.

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs (hunk 2 above)
Comment (smelc) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs:1460]: Follow-up issue created: https://github.com/IntersectMBO/cardano-cli/issues/606

**Pull request merged**